sdk/core/core-https/src/interfaces.ts

```diff
@@ -274,3 +274,18 @@ export type FormDataValue = string | Blob;
  * A simple object that provides form data, as if from a browser form.
  */
 export type FormDataMap = { [key: string]: FormDataValue | FormDataValue[] };
+
+/**
+ * Allows the discovery of authentication properties.
+ */
+export interface AuthenticationContext {
```

Comment:
should this live in `core-auth` maybe?

Reply:
This is really specific to the challenge methods in bearerTokenAuthenticationPolicy. We could also name it like `BearerTokenProcessChallengeResponse`. We could also inline it like this: `processChallenge(challenge: string): Promise<{ scopes?: string[], claims?: string } | undefined>;` to avoid having to have a new interface. 🤔

Reply:
I'll go with `BearerTokenChallengeResult` for now, but please feel free to point to another direction!